backend/LexBoxApi/Services/RepoMigrationService.cs

```diff
@@ -34,6 +35,40 @@ public void QueueMigration(string projectCode)
         ProjectMigrationQueue.Writer.TryWrite(projectCode);
     }
 
+    /// <summary>
+    /// awaits until the project has been migrated, or the token is cancelled
+    /// </summary>
+    public async Task WaitMigrationFinishedAsync(string projectCode, CancellationToken cancellationToken)
+    {
+        var channel = Channel.CreateUnbounded<string>();
+        var key = Guid.NewGuid();
+        using var cleanup = Defer.Action(() =>
+        {
+            channel.Writer.TryComplete();
+            _projectMigrationChannels.TryRemove(key, out _);
+        });
+        _projectMigrationChannels.TryAdd(key, channel);
+        //only way this stops is if the token is cancelled.
+        await foreach (var projectMigrated in channel.Reader.ReadAllAsync(cancellationToken))
+        {
+            if (projectMigrated != projectCode) continue;
```

Comment:
Why don't we just use the project-code as the key and only write to the relevant channel?

Reply:
because we could have more than 1 listener for each project code.